src/net.h

```diff
@@ -251,7 +254,12 @@ class CConnman
     void SetServices(const CService &addr, ServiceFlags nServices);
     void MarkAddressGood(const CAddress& addr);
     void AddNewAddresses(const std::vector<CAddress>& vAddr, const CAddress& addrFrom, int64_t nTimePenalty = 0);
-    std::vector<CAddress> GetAddresses();
+
+    // Cache is used to minimize topology leaks, so it should
+    // be used for all non-trusted calls, for example, p2p.
+    // NET_INTERNAL represents a non-malicious call (either from RPC) or
```

Comment:
I think this should probably be a whitelist flag, not an assumption internal nodes are non-malicious